components/inventory.ts

```diff
@@ -547,22 +547,24 @@ export function getUnlockablesById(
  * @param profileId The profileId of the player
  * @param gameVersion The game version
  * @param inv The inventory to update
- * @param sublocation The sublocation to check for a default suit
+ * @param sublocation The sublocation to check for a default suit.
+ * @param suitOverride A default suit override for this contract.
  * @returns The updated inventory
  */
 function updateWithDefaultSuit(
     profileId: string,
     gameVersion: GameVersion,
     inv: InventoryItem[],
     sublocation?: Unlockable,
+    suitOverride?: string,
 ): InventoryItem[] {
     if (!sublocation) {
         return inv
     }
 
     // Yes this is slow. We should organize the unlockables into a { [Id: string]: Unlockable } map.
     const locationSuit = getUnlockableById(
-        getDefaultSuitFor(sublocation),
+        getDefaultSuitFor(sublocation, suitOverride),
```

Comment:
I think it would be better if this were something like `suitOverride && getUnlockableById(suitOverride) ?? getUnlockableById(getDefaultSuitFor(sublocation))` so we can fall back to the location default for contracts with invalid values here